### PR TITLE
fix: start background processing only if the app started

### LINKED ISSUE #1252: Background jobs are started despite application not being ready
There are multiple errors on starting the app from scratch.
Boutique app does not start because the background job is trying to query the db that is not yet ready.
Same in wallet - there are multiple queries to db to process pending transactions. The wallet is eventually starting after the connection to db is established.

The background jobs should start only if the application started successfully.
image

### PR DESCRIPTION
**Pull request by dragosp1011**
### Context

- fixes #1252

### CODE REVIEW

File: packages/boutique/backend/src/index.ts [hunk 1]
```diff
@@ -9,10 +11,14 @@ export const start = async (app: App): Promise<void> => {
 async function bootstrap() {
   const container = await createContainer(env)
   const app = new App(container)
-  await start(app).catch(async (e): Promise<void> => {
+  try {
+    await start(app)
+    await app.processResources()
+  } catch (e) {
+    console.log('Error on starting the app')
```
Comment (Tymmmy) [inline packages/boutique/backend/src/index.ts:18]: should we leave in console logs?

Comment (dragosp1011) [reply packages/boutique/backend/src/index.ts:18]: yes, the app is not initialised on this step yet

File: packages/boutique/backend/src/app.ts [hunk 2]
```diff
@@ -132,7 +132,7 @@ export class App {
       .catch((err) => {
         logger.error('Error while trying to process pending payments')
         logger.error(err)
-        return true
+        return false
```
Comment (beniaminmunteanu) [inline packages/boutique/backend/src/app.ts:135]: why is the return needed at all?

Comment (dragosp1011) [reply packages/boutique/backend/src/app.ts:135]: yes, to setup the next execution

Comment (dragosp1011) [reply packages/boutique/backend/src/app.ts:135]: with returning true, it was going inside the if with immediate execution and flooding the app with errors

**Review by beniaminmunteanu — APPROVED** [review]

**Review by Tymmmy — APPROVED** [review]

**Pull request merged**